bakery/checkers/checkers.go

```diff
@@ -235,3 +237,48 @@ func ClientIPAddrCaveat(addr net.IP) Caveat {
 func ErrorCaveatf(f string, a ...interface{}) Caveat {
 	return firstParty(CondError, fmt.Sprintf(f, a...))
 }
+
+// AllowCaveat returns a caveat that will deny attempts to use the
+// macaroon to perform any operation other than those listed.
+func AllowCaveat(op ...string) Caveat {
+	return firstParty(CondAllow, strings.Join(op, " "))
+}
+
+// DenyCaveat returns a caveat that will deny attempts to use the
+// macaroon to perform any of the listed operations.
+func DenyCaveat(op ...string) Caveat {
+	return firstParty(CondDeny, strings.Join(op, " "))
+}
+
+// OperationChecker checks any allow or deny caveats ensuring they do not
+// prohibit the named operation.
+type OperationChecker string
+
+// Condition implements Checker.Condition.
+func (OperationChecker) Condition() string {
+	return ""
+}
+
+// Check implements Checker.Check.
+func (o OperationChecker) Check(cond, arg string) error {
```

Comment:
alternative:

	func (o OperationChecker) Check(cond, arg string) error {
	    found := ""
	    for _, op := range strings.Fields(arg) {
	        if string(o) == op {
	             found = op
	        }
	    }
	    switch cond {
	    case CondAllow:
	        if found == "" {
	             return errgo.Newf("operation %s not allowed", o)
	        }
	    case CondDeny:
	       if found != "" {
	            return errgo.Newf("operation %s not allowed", found)
	       }
	    default:
	        return ErrCaveatNotRecognized
	    }
	    return nil
	}

?